src/xrandrrutil.cpp

```diff
@@ -45,14 +45,18 @@ unsigned int refreshFromModeInfo(const XRRModeInfo &modeInfo) {
     return static_cast<unsigned int>(round(rate));
 }
 
-const string renderXrandrCmd(const list<shared_ptr<Output>> &outputs, const shared_ptr<Output> &primary, const long &dpi) {
+const string renderXrandrCmd(const list<shared_ptr<Output>> &outputs, const shared_ptr<Output> &primary, const long &dpi, const long &rate) {
     stringstream ss;
     ss << "xrandr \\\n --dpi " << dpi;
     for (const auto &output : outputs) {
         ss << " \\\n --output " << output->name;
         if (output->desiredActive && output->desiredMode && output->desiredPos) {
             ss << " --mode " << output->desiredMode->width << "x" << output->desiredMode->height;
-            ss << " --rate " << output->desiredMode->refresh;
+            if (rate == 0) {
+                ss << " --rate " << output->desiredMode->refresh;
+            } else {
+                ss << " --rate " << rate;
```

Comment:
This applies the user provided refresh rate to all monitors.

That is OK (at least on my system) as it will result in a noop if the rate is invalid.

Reply:
Xrandr handle wrong refresh rate very good. I tested several values, and if one enters wrong refresh rate, xrandr finds the closest refresh rate. The implementation of different rate for different output, is more complex to implement, since it has to be done similar like order of screens feature.